coderd/workspaceagents.go

```diff
@@ -900,6 +904,60 @@ func (api *API) workspaceAgentReportStats(rw http.ResponseWriter, r *http.Reques
 	})
 }
 
+// @Summary Submit workspace agent lifecycle state
+// @ID submit-workspace-agent-lifecycle-state
+// @Security CoderSessionToken
+// @Accept json
+// @Tags Agents
+// @Param request body codersdk.PostWorkspaceAgentLifecycleRequest true "Workspace agent lifecycle request"
+// @Success 204 "Success"
+// @Router /workspaceagents/me/report-lifecycle [post]
```

Comment:
nit: if you want to hide this endpoint from docs, you can add: `// @x-apidocgen {"skip": true}`

Reply:
Fixed!